queries/orderbook/protocol_fees.sql

```diff
@@ -0,0 +1,105 @@
+-- Active: 1704993022452@@127.0.0.1@5432@postgres@public
+-- Active: 1704193756200@@cow-protocol-db-read-replica.c5bze0gkehvb.eu-central-1.rds.amazonaws.com@5432@mainnet@public
+WITH order_surplus AS (
+    SELECT
+        at.auction_id,
+        s.tx_hash,
+        t.order_uid,
+        o.sell_token,
+        o.buy_token,
+        t.sell_amount,
+        t.buy_amount,
+        oe.surplus_fee,
+        o.kind,
+        CASE
+            WHEN o.kind = 'sell'
+                THEN t.buy_amount - o.buy_amount / (o.sell_amount + o.fee_amount) * t.sell_amount
+            WHEN o.kind = 'buy'
+                THEN (o.sell_amount + o.fee_amount) / o.buy_amount * t.buy_amount - t.sell_amount
+        END AS surplus,
+        CASE
+            WHEN o.kind = 'sell'
+                THEN o.buy_token
+            WHEN o.kind = 'buy'
+                THEN o.sell_token
+        END AS surplus_token,
+        CASE
+            WHEN o.kind = 'sell'
+                THEN
+                    t.buy_amount - oq.buy_amount / (o.sell_amount + o.fee_amount) * t.sell_amount
+            WHEN o.kind = 'buy'
+                THEN
+                    (oq.sell_amount + o.fee_amount) / o.buy_amount * t.buy_amount - t.sell_amount
+        END AS price_improvement
```

Comment:
I will probably remove this since the initial implementation will only consider surplus and not price improvement. This should simplify the code a bit.